pandas/tests/arithmetic/test_interval.py

```diff
@@ -126,12 +126,15 @@ def test_compare_scalar_interval_mixed_closed(self, op, closed, other_closed):
         expected = self.elementwise_comparison(op, array, other)
         tm.assert_numpy_array_equal(result, expected)
 
-    def test_compare_scalar_na(self, op, array, nulls_fixture):
+    def test_compare_scalar_na(self, op, array, nulls_fixture, request):
         result = op(array, nulls_fixture)
         expected = self.elementwise_comparison(op, array, nulls_fixture)
 
         if nulls_fixture is pd.NA and array.dtype != pd.IntervalDtype("int"):
-            pytest.xfail("broken for non-integer IntervalArray; see GH 31882")
+            mark = pytest.mark.xfail(
```

Comment:
Perhaps a helpful function for this if we are going to do across the codebase? Jeff's idea - [here](https://github.com/pandas-dev/pandas/pull/30521#issuecomment-569676489)

I'm happy to tackle some myself too :)

Reply:
Go for it.  I think this is a strictly better pattern than `pytest.xfail`, since it will alert us if an xfail becomes fixed.